virocon/distributions.py

```diff
@@ -1184,3 +1185,163 @@ def _wlsq_error(delta, x, p, w, return_alpha_beta=False, falpha=None, fbeta=None
         wlsq_error = np.sum(w * (x - x_hat) ** 2)
 
         return wlsq_error
+
+
+class GeneralizedGammaDistribution(Distribution):
+    """
+    A generalized Gamma distribution. 
+   
+    The distributions probability density function is given by [5]_ . 
+    The parametrization is orientated on [5]_ :
+    
+    :math:`f(x) = \\frac{ \\lambda^{cm} cx^{cm-1} \\exp \\left[ - \\left(\\lambda x^{c} \\right) \\right] }{\\Gamma(m)}`
+    
+    Parameters
+    ----------
+    m : float
+        Shape parameter of the generalized Gamma distribution. Defaults to 1.
+    c : float
+        Second shape parameter of the generalized Gamma distribution. Defaults
+        to 1.
+    lambda_ : float
+        Additional scale parameter of the generalized Gamma distribution 
```

Comment:
Is lambda_ really a scale parameter? This might depend on our definition of a scale parameter. If we use scipy's definition this is not a scale parameter (according to scipy, scale is used in the form: (x -loc) / scale).
Maybe we could name it reciprocal scale parameter?

Reply:
Good point. I looked it up in the book "Statistical distributions" by Forbes et. al (2011). Here, b is defined as the scale parameter and lambda_ is just the inverse of b. @ahaselsteiner what do you think?

Reply:
Lambda looks like a scale parameter to me. It scales the distribution. However, why is it an "additional" scale parameter? I think "additional" can be deleted. It's the only one.

I think for naming parameters and for the concept of how location and scale are implemented, scipy is not a good reference. They tried to "harmonize" a bit too much such they sometimes could not stick to standard parameterizations. For theory, and our implementations we should stick to the academic literature.

Reply:
Well if Forbes et. al (2011) also names b the scale parameter and lambda_ is the inverse, I think we could stick to that. 
You are right that the lambda_ also scales the distribution but it does so inversely to e.g. b. 
Say we agree that sigma is the scale parameter for a normal distribution. If sigma grows the distribution becomes more spread and more narrow if sigma shrinks. To me this could be 'the' definition for a scale parameter. 
Thus I would prefer reciprocal scale for lambda_.

Reply:
This is a screenshot of Forbes et. al (2011). I do not have a strong opinion on that, but maybe sticking to 'scale' is more consistent and understandably for the user. 

<img width="487" alt="Bildschirmfoto 2021-11-22 um 11 35 02" src="https://user-images.githubusercontent.com/57865209/142846490-55103eb7-bcb0-4ca1-afc5-cb1f635a849c.png">

Reply:
I think it is more consistent to use scale when we actually have a scale parameter. I also assume that reciprocal scale will not confuse users. Maybe name it (reciprocal) scale, to emphasize that lambda_ is sort of a scale parameter?

Reply:
I think it is a good idea to name it reciprocal_scale!